packages/core/src/__types__/base.ts

```diff
@@ -0,0 +1,528 @@
+  };
+};
+
+export interface ApiKeyStamperBase {
+  listKeyPairs(): Promise<string[]>;
+  createKeyPair(
+    externalKeyPair?: CryptoKeyPair | { publicKey: string; privateKey: string },
+  ): Promise<string>;
+  deleteKeyPair(publicKeyHex: string): Promise<void>;
+  clearKeyPairs(): Promise<void>;
+  stamp(payload: string, publicKeyHex: string): Promise<TStamp>;
+}
+
+export interface WalletProviderInfo {
+  name: string;
+  uuid?: string;
+  icon?: string;
+  rdns?: string;
+}
+
+export enum Chain {
+  Ethereum = "ethereum",
+  Solana = "solana",
+}
+
+export type SwitchableChain = {
+  id: string;
+  name: string;
+  rpcUrls: string[];
+  blockExplorerUrls?: string[];
+  iconUrls?: string[];
+  nativeCurrency: {
+    name: string;
+    symbol: string;
+    decimals: number;
+  };
+};
+
+export type EvmChainInfo = {
+  namespace: Chain.Ethereum;
+  chainId: string;
+};
+
+export type SolanaChainInfo = {
+  namespace: Chain.Solana;
+};
+
+export type ChainInfo = EvmChainInfo | SolanaChainInfo;
+
+export interface WalletConnectProvider {
+  request(args: { method: string; params?: any[] }): Promise<unknown>;
+}
+
+export type WalletRpcProvider =
+  | EthereumProvider
+  | SolanaProvider
+  | WalletConnectProvider;
+
+export interface WalletProvider {
+  interfaceType: WalletInterfaceType;
+  chainInfo: ChainInfo;
+  info: WalletProviderInfo;
+  provider: WalletRpcProvider;
+  connectedAddresses: string[];
+  uri?: string;
+}
+
+export enum SignIntent {
+  SignMessage = "sign_message",
+  SignTransaction = "sign_transaction",
+  SignAndSendTransaction = "sign_and_send",
+}
+
+export enum WalletInterfaceType {
+  Solana = "solana",
+  Ethereum = "ethereum",
+  WalletConnect = "wallet_connect",
+}
+
+/**
+ * Base interface for wallet functionalities common across different blockchain chains.
+ * @interface BaseWalletInterface
+ * @property {function(string): Promise<string>} signMessage - Signs a message and returns the hex signature as a string.
+ * @property {function(): Promise<string>} getPublicKey - Retrieves the public key as a string.
+ */
+export interface BaseWalletInterface {
+  interfaceType: WalletInterfaceType;
+  sign: (
+    message: string,
+    provider: WalletProvider,
+    intent: SignIntent,
+  ) => Promise<string>;
+  getPublicKey: (provider: WalletProvider) => Promise<string>;
+  getProviders: () => Promise<WalletProvider[]>;
+  connectWalletAccount: (provider: WalletProvider) => Promise<void>;
+  disconnectWalletAccount: (provider: WalletProvider) => Promise<void>;
+  switchChain?: (
+    provider: WalletProvider,
+    chainOrId: string | SwitchableChain,
+  ) => Promise<void>;
+}
+
+/**
+ * Solana wallets can directly access the public key without needing a signed message.
+ * @interface SolanaWalletInterface
+ * @extends BaseWalletInterface
+ * @property {function(): string} getPublicKey - Returns the public key, which is the ED25519 hex encoded public key from your Solana wallet public key.
```

Comment:
tiny nit here and in a few other places in this file: it might be more accurate to say `hex-encoded Ed25519 public key`